app/src/main/java/to/bitkit/repositories/ActivityRepo.kt

```diff
@@ -330,6 +322,79 @@ class ActivityRepo @Inject constructor(
         }
     }
 
+    private suspend fun syncTagsMetaData(
+    ) = withContext(context = bgDispatcher) {
+        runCatching {
+            if (db.tagMetadataDao().getAll().isEmpty()) return@withContext
+            val lastActivities = getActivities(limit = 10u).getOrNull() ?: return@withContext
+            Logger.debug("syncTagsMetaData called")
+            lastActivities.forEach { activity ->
```

Comment:
Computation inside `forEach` block should be parallelized later, when optimising for performance.